rokka/sourceimages.go

```diff
@@ -122,3 +126,36 @@ func (c *Client) CreateSourceImageWithMetadata(org, name string, data io.Reader,
 
 	return result, err
 }
+
+func (c *Client) DynamicMetadataResponseHandler(v interface{}, resp *http.Response) error {
+	if resp.StatusCode == 204 {
+		dynamicMetadataResponse, ok := v.(*PutDynamicMetadataResponse)
+
+		if ok {
+			dynamicMetadataResponse.Location = resp.Header.Get("Location")
+
+			return nil
+		}
+	}
+
+	return StatusCodeError{
+		resp.StatusCode,
+		[]byte{},
+	}
+}
+
+func (c *Client) AddDynamicMetadata(org, hash string, name string, jsonBody string) (PutDynamicMetadataResponse, error) {
+	result := PutDynamicMetadataResponse{}
+
+	b := bytes.NewBuffer([]byte(jsonBody))
```

Comment:
[`bytes.NewBufferString`](https://godoc.org/bytes#NewBufferString) should be used instead.